### ISSUE TITLE
feat(distributions): conditional Gaussian mixture with full Cholesky precision

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/gauss_flows#71]
## Problem / Request
Add a conditional Gaussian mixture distribution (`ConditionalGaussianMixture`) with full per-component Cholesky-factored precision matrices. This is the MDN-style base distribution used in Ardizzone's cINN work — distinct from the 1D elementwise `MixtureGaussianCDF` / `MixtureLogisticCDF` already in gauss_flows.

## User Story
> As a researcher training conditional INNs (cINN) or mixture-density networks for multimodal inverse problems, I want a first-class conditional GMM distribution with full-covariance components, so that I can use it as a flexible base distribution under a flow or as a standalone MDN without reimplementing log-prob, sampling, and numerical-stability tricks.

## Motivation
- **Flexible base.** A conditional GMM base composed under a flow gives multimodal latents conditional on context — the backbone of cINN for inverse problems.
- **Multimodal regression.** MDNs directly target multimodal `p(y | x)` (e.g. Ardizzone 2018 inverse-problems paper).
- **Full covariance.** Per-component Cholesky gives correlated latent directions; the existing `MixtureGaussianCDF` is only 1D elementwise.
- **Ties into gaussx.** Shares linear-algebra primitives (triangular solves, logdet) with the gaussx `MultivariateNormal` work — good reuse opportunity.

## Proposed API
```python
import equinox as eqx
from jax import Array

class ConditionalGaussianMixture(eqx.Module):
    """Conditional GMM p(x | c) = Σ_k π_k(c) N(x; μ_k(c), Σ_k(c)).

    Each component has a full covariance parameterised by its Cholesky
    factor of the *precision* matrix (lower triangular L with positive
    diagonal): Σ_k^{-1} = L_k L_k^T. log-prob, sampling, and mean/mode
    all use L_k directly for numerical stability.

    Args:
        context_to_params: eqx.Module mapping context c -> (logits_pi,
            means, log_diags, off_diags), where the latter three are
            reshaped into the Cholesky precision factors per component.
        n_components: K.
        dim: Data dimension d.
        cond_dim: Context dimension.
    """
    context_to_params: eqx.Module
    n_components: int = eqx.field(static=True)
    dim: int = eqx.field(static=True)
    cond_dim: int = eqx.field(static=True)

    def log_prob(self, x: Array, c: Array) -> Array: ...
    def sample(self, key, c: Array, shape: tuple = ()) -> Array: ...
    def nll_upper_bound(self, x: Array, c: Array) -> Array:
        # Jensen upper bound: -Σ_k π_k log N(x; μ_k, Σ_k)
        # Cheaper than log-sum-exp; used as a training surrogate in cINN.
        ...
```

## Design Snapshot
```python
# src/gauss_flows/_src/distributions/conditional_gmm.py
import jax
import jax.numpy as jnp
import equinox as eqx
from jax.nn import log_softmax

def _tril_from_flat(log_diag, off_diag, d):
    # Build lower-triangular L (d, d) with positive diagonal (softplus)
    # from a flat parameterisation.
    L = jnp.zeros((d, d))
    L = L.at[jnp.arange(d), jnp.arange(d)].set(jax.nn.softplus(log_diag) + 1e-5)
    i, j = jnp.tril_indices(d, k=-1)
    L = L.at[i, j].set(off_diag)
    return L

def _component_log_prob(x, mu, L):
    # log N(x; mu, (L L^T)^{-1})  where L is the lower-Cholesky of the
    # *precision*:
    #   log p = -0.5 d log(2π) + Σ log(diag L) - 0.5 ||L^T (x - mu)||^2
    z = L.T @ (x - mu)
    d = x.shape[-1]
    return (-0.5 * d * jnp.log(2 * jnp.pi)
            + jnp.sum(jnp.log(jnp.diag(L)))
            - 0.5 * jnp.sum(z * z))

class ConditionalGaussianMixture(eqx.Module):
    context_to_params: eqx.Module
    n_components: int = eqx.field(static=True)
    dim: int = eqx.field(static=True)
    cond_dim: int = eqx.field(static=True)

    def _components(self, c):
        raw = self.context_to_params(c)          # flat params vector
        # split raw → logits_pi (K,), means (K, d), log_diags (K, d),
        # off_diags (K, d*(d-1)/2)
        ...
        Ls = jax.vmap(_tril_from_flat, in_axes=(0, 0, None))(log_diags, off_diags, self.dim)
        return log_softmax(logits_pi), means, Ls

    def log_prob(self, x, c):
        log_pi, mus, Ls = self._components(c)    # (K,), (K,d), (K,d,d)
        lps = jax.vmap(_component_log_prob, in_axes=(None, 0, 0))(x, mus, Ls)
        return jax.scipy.special.logsumexp(log_pi + lps)

    def sample(self, key, c, shape=()):
        log_pi, mus, Ls = self._components(c)
        k1, k2 = jax.random.split(key)
        k = jax.random.categorical(k1, log_pi, shape=shape)
        eps = jax.random.normal(k2, shape + (self.dim,))
        # Σ_k = (L_k L_k^T)^{-1} ⇒ sample via solve_triangular(L^T, eps, lower=False)
        L_k = Ls[k]
        x = mus[k] + jax.scipy.linalg.solve_triangular(
            jnp.swapaxes(L_k, -1, -2), eps[..., None], lower=False
        ).squeeze(-1)
        return x
```

## Mathematical Notes
```text
Per component k:
    precision Λ_k = L_k L_k^T        (L_k lower triangular, diag > 0)
    cov       Σ_k = Λ_k^{-1}

log N(x; μ_k, Σ_k) = -(d/2) log(2π) + Σ_j log L_k[j,j] - ½ ||L_k^T (x - μ_k)||²

Full log-prob:
    log p(x | c) = logsumexp_k [ log π_k(c) + log N(x; μ_k(c), Σ_k(c)) ]

Sampling x ~ N(μ, Λ^{-1}):
    solve L^T z = ε  (triangular, upper)  →  x = μ + z
    where ε ~ N(0, I). Note L^T is upper triangular; use
    jax.scipy.linalg.solve_triangular(L^T, ε, lower=False).

Stability: parameterise L's diagonal via softplus + floor to keep strictly
positive; clip off-diagonal magnitudes if needed.
```

## References & Existing Code
- Papers:
  - Ardizzone et al. 2019. *Guided Image Generation with Conditional Invertible Neural Networks*. arXiv:1907.02392 (cINN).
  - Bishop 1994. *Mixture Density Networks*. (original MDN.)
- Reference impl: `vislearn/FrEIA/FrEIA/modules/gaussian_mixture.py::GaussianMixtureModel`
- gauss_flows analogues to coordinate with:
  - `src/gauss_flows/_src/distributions/mixture.py` (if it exists — check)
  - `src/gauss_flows/_src/transforms/bijections/elementwise/mixture_cdf.py` (1D CDFs — not a replacement)
- Cross-repo: gaussx `MultivariateNormal` — share Cholesky / triangular-solve conventions.

## Implementation Steps
- [ ] Add `ConditionalGaussianMixture` in `src/gauss_flows/_src/distributions/conditional_gmm.py`
- [ ] Re-export from `src/gauss_flows/_src/distributions/__init__.py` and top-level `gauss_flows`
- [ ] Helper `context_to_params` factory (simple MLP wrapper) so users don't have to build the param-splitter by hand
- [ ] Add `nll_upper_bound` method (Jensen bound) — optional training surrogate used in FrEIA

## Definition of Done
- [ ] Code at intended path
- [ ] Public API exported
- [ ] Tests pass
- [ ] Lint + typecheck pass
- [ ] Google-style docstrings

## Testing
- [ ] `log_prob` matches a reference scipy/numpy implementation on a handcrafted (K=2, d=3) example
- [ ] `sample` + KDE → density roughly recovers `log_prob` (statistical)
- [ ] Component reduction: K=1 → matches plain full-cov Gaussian log-prob
- [ ] Jensen bound satisfies `nll_upper_bound ≥ -log_prob`
- [ ] Gradient flow: `jax.grad` of `log_prob` w.r.t. context produces finite values

## Documentation
- [ ] API reference entry
- [ ] Short usage recipe in notebooks showing cINN-style multimodal regression

## Relationships
- Related: FrEIA port series; future integration with gaussx `MultivariateNormal`
